install_sct

```diff
@@ -717,7 +717,7 @@ conda deactivate >/dev/null 2>&1
 if [[ -e "$RC_FILE_PATH" ]]; then
     if grep "sct_env" "$RC_FILE_PATH"; then
       print info "In case an old version SCT is already installed (4.0.0-beta.1 or before), remove 'sct_env' declaration in RC file"
-      perl -pi -e '/sct_env/ s/^#*/#/' "$RC_FILE_PATH"
+      perl -pi -e 's/^(sct_env)/# \1/' "$RC_FILE_PATH"
```

Comment:
> I fear this will be difficult to test, because I've never seen a pre-4.0 SCT-installed bashrc file.

[Prior to v4.0.0](https://github.com/spinalcordtoolbox/spinalcordtoolbox/commit/3db47876c08ca60af24a3a88ae798aab87c6cf36), the line in `install_sct` to add `sct_env` was:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/3db47876c08ca60af24a3a88ae798aab87c6cf36/install_sct#L93

<details>
<summary> Aside: sct_env </summary>

> _For historical context, `sct_env` itself was a bash script that set other environment variables:_
> https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/3db47876c08ca60af24a3a88ae798aab87c6cf36/install/sct_env#L1-L3

</details>

So, we should be able to test the perl command manually by running:

```console
$ echo source /home/joshua/repos/spinalcordtoolbox/bin/sct_env > test_file.txt

$ cat test_file.txt
source /home/joshua/repos/spinalcordtoolbox/bin/sct_env

$ perl -pi -e '/sct_env/ s/^#*/#/' test_file.txt
syntax error at -e line 1, near "/sct_env/ s/#*/#/"
Execution of -e aborted due to compilation errors.

$ cat test_file.txt
source /home/joshua/repos/spinalcordtoolbox/bin/sct_env

$ perl -pi -e 's/^(sct_env)/# *\1/' test_file.txt

$ cat test_file.txt
source /home/joshua/repos/spinalcordtoolbox/bin/# sct_env
```

I think the previous `sed` command was attempting to instead prepend `#` to the line:

```console
$ sed -i '/sct_env/ s/^#*/#/' test_file.txt

$ cat test_file.txt
#source /home/joshua/repos/spinalcordtoolbox/bin/sct_env
```

So, here's a proposed update, which includes the rest of the line [in the regex capturing group](https://regexr.com/6mv82):

```suggestion
      perl -pi -e 's/^(.*sct_env.*)$/# \1/' "$RC_FILE_PATH"
```

```console
$ perl -pi -e 's/^(.*sct_env.*)$/# \1/' test_file.txt

$ cat test_file.txt
# source /home/joshua/repos/spinalcordtoolbox/bin/sct_env
```

Reply:
Hm, this form isn't idempotent though:

```
parker:~ p115628$ echo source /home/joshua/repos/spinalcordtoolbox/bin/sct_env > test_file.txt
parker:~ p115628$ perl -pi -e 's/^(.*bin\/sct_env)/# \1/' test_file.txt 
parker:~ p115628$ cat test_file.txt 
# source /home/joshua/repos/spinalcordtoolbox/bin/sct_env
parker:~ p115628$ perl -pi -e 's/^(.*bin\/sct_env)/# \1/' test_file.txt 
parker:~ p115628$ cat test_file.txt 
# # source /home/joshua/repos/spinalcordtoolbox/bin/sct_env
parker:~ p115628$ perl -pi -e 's/^(.*bin\/sct_env)/# \1/' test_file.txt 
parker:~ p115628$ cat test_file.txt 
# # # source /home/joshua/repos/spinalcordtoolbox/bin/sct_env
```

Hang on, I'm sure I can fix that...

Reply:
This seems to work:

```suggestion
      perl -pi -e 's/^(# )?(.*bin/sct_env)/# \2/' "$RC_FILE_PATH"
```

```
parker:~ p115628$ echo source /home/joshua/repos/spinalcordtoolbox/bin/sct_env > test_file.txt
parker:~ p115628$ perl -pi -e 's/^(# )?(.*bin\/sct_env)/# \2/' test_file.txt 
parker:~ p115628$ cat test_file.txt 
# source /home/joshua/repos/spinalcordtoolbox/bin/sct_env
parker:~ p115628$ perl -pi -e 's/^(# )?(.*bin\/sct_env)/# \2/' test_file.txt 
parker:~ p115628$ cat test_file.txt 
# source /home/joshua/repos/spinalcordtoolbox/bin/sct_env
parker:~ p115628$ perl -pi -e 's/^(# )?(.*bin\/sct_env)/# \2/' test_file.txt 
parker:~ p115628$ cat test_file.txt 
# source /home/joshua/repos/spinalcordtoolbox/bin/sct_env
```

Reply:
Ahhh, very good catch about running it multiple times.